meson.build

```diff
@@ -0,0 +1,299 @@
+
+version_array = version_stable.split('.')
+version_major = version_array[0]
+version_minor = version_array[1]
+version_micro = version_array[2]
+
+# The API "platform" or "intercompatibility" version.
+#
+# This one is used for library name prefixes, for introspection
+# namespace versions, for gettext domains, and basically anything that
+# needs to change when backwards or forwards API compatibility changes.
+# Another way of thinking about it: it allows meaningful side by side
+# installations of libmypaint.
+api_platform_version = f'@version_major@.@version_minor@'
+api_name = f'libmypaint-@api_platform_version@'
+
+project_url = 'https://github.com/mypaint/libmypaint'
+
+conf.set('PACKAGE_NAME', meson.project_name())
+conf.set('PACKAGE_URL', project_url)
+conf.set('LIBMYPAINT_API_PLATFORM_VERSION', api_platform_version)
+conf.set('LIBMYPAINT_VERSION', version_stable)
+conf.set('LIBMYPAINT_VERSION_FULL', version_full)
+
+gettext_package = api_name
+conf.set_quoted(
+  'GETTEXT_PACKAGE',
+  gettext_package,
+  description: 'The prefix for our gettext translation domains.',
+)
+
+###############################################################################
+# Libtool versioning
+
+# ABI version see: https://autotools.io/libtool/version.html
+# https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
+abi_revision = 0 # increment on every release
+abi_current = 0 # inc when add/remove/change interfaces
+abi_age = 0 # inc only if changes backward compat
+abi_version_info = f'@abi_current@.@abi_revision@.@abi_age@'
+
+###############################################################################
+# System detection, compiler options
+
+platform_win32 = (host_machine.system() == 'windows')
+platform_osx = (host_machine.system() == 'darwin')
+
+# Define strdup() in string.h under glibc >= 2.10 (POSIX.1-2008)
+add_project_arguments('-D_POSIX_C_SOURCE=200809L', language: 'c')
+
+###############################################################################
+# Dependencies
+
+libmath = cc.find_library('m', required: false)
+
+json = dependency('json-c')
+
+# glib
+gobject = dependency('gobject-2.0', required: get_option('glib'))
+use_glib = gobject.found()
+conf.set10('MYPAINT_CONFIG_USE_GLIB', use_glib)
+
+# GEGL
+gegl = dependency('gegl-0.4', 'gegl-0.3', required: get_option('gegl'))
+use_gegl = gegl.found()
+if use_gegl
+  gegl_gir = gegl.version().version_compare('>=0.4') ? 'Gegl-0.4' : 'Gegl-0.3'
+endif
+
+introspection_required_version = '1.32.0'
+introspection_feature = get_option(
+  'introspection',
+).require(
+  use_glib,
+  error_message: 'Generating GObject introspection requires building with GLib support',
+)
+# For g-ir-scanner and g-ir-compiler as used by the gnome module.
+gi = dependency(
+  'gobject-introspection-1.0',
+  version: f'>=@introspection_required_version@',
+  required: introspection_feature,
+  native: true,
+)
+use_introspection = gi.found()
```

Comment:
Not a fan of unused dependency like this but if we just use boolean defaulting to `true`, it will fail due to missing `glib` when `auto_features=disabled`. And to make `auto` feature work as expected, we need to pass it to `dependency` (unfortunately, `gnome.generate_gir` does not have `required` kwarg).